### ISSUE TITLE
Run (not install) a centrilized Waypoint server and reach it from a remote workstation 

### ISSUE THREAD

**Issue by mreferre** [hashicorp/waypoint#1217]
I am opening this as a feature request because I am not even sure this is something WP even support. 

I would like to run a server (i.e. not install it) on an EC2 instance running Amazon Linux and I would like to connect to this remote Waypoint server from a workstation. 

I have tried a couple of routes. The first one is the classic commands suggested in the docs: 
```
waypoint server run -accept-tos -db=/waypoint/waypointdb.db &
waypoint server bootstrap -server-addr=localhost:9701 -server-tls-skip-verify
``` 
This works fine locally and I can `waypoint context verify` properly. 

If I now move to my remote workstation and I try to create a context to point to this server, my context is created just fine (`1.2.3.4` and `abcdefgh` are dummy placeholders that represent the public ip of my server and the WP token):
```
waypoint context create -server-addr=1.2.3.4:9701 -server-auth-token=abcdefgh -server-tls-skip-verify -set-default my-server
Context "my-server" created.
```
However, it fails to verify:
```
$ waypoint context verify
 Connecting with context "my-server"...
! Error connecting with context "my-server": context deadline exceeded
``` 

One of my hypothesis was that the waypoint server was only bound to `localhost` so I tried to recreate the server bootstrapping it with the public ip of the EC2 instance but the bootstrap just times out and then fails on "client creation": 
```
waypoint server run -accept-tos -db=/waypoint/waypointdb.db &
waypoint server bootstrap -server-addr=1.2.3.4:9701 -server-tls-skip-verify
! failed to create client: context deadline exceeded
``` 

It goes without saying that the EC2 instance has a WIDE open security group (for short experimentation) that would let all incoming traffic flow. 

To recap, I would like to be able to run (not install) the waypoint server on a centralized instance and create a context that points to it from a remote workstation. The fact all docs suggest to bind the server to localhost | 127.0.0.1 makes me feel this is not even supported? 

Thanks.

**Comment by mreferre** [comment]
@nicholasjackson to the rescue! I eventually found [this article](https://discuss.hashicorp.com/t/how-to-install-waypoint-server-to-azure/16462) and Nic's suggestions were on point. 

For posterity, I did start my server on the EC2 instance like this: 
```
PUBLIC_IP=$(curl http://169.254.169.254/latest/meta-data/public-ipv4)
waypoint server run -accept-tos -db=/waypoint/waypointdb.db \
        -listen-grpc=0.0.0.0:9701 \
        -listen-http=0.0.0.0:9702 \
        -advertise-addr=$PUBLIC_IP:9701 \
        -advertise-tls=true \
        -advertise-tls-skip-verify=true &
        
waypoint server bootstrap -server-addr=0.0.0.0:9701 -server-tls-skip-verify
```
And I created the context on the remote workstation like this: 
```
waypoint context create -server-addr=1.2.3.4:9701 -server-auth-token=abcdefg -server-tls-skip-verify -set-default my-server
```

And this time it worked out just fine: 
```
$ waypoint context verify
✓ Context "my-server" connected successfully.
```

**Comment by briancain** [comment]
Hey @mreferre ! Glad you found a solution, and appreciate the follow up on how you solved it 😄 Thanks for giving Waypoint a shot!

**Comment by O4epegb** [comment]
Also had this issue (used `waypoint install` to start the server though),

```
$ waypoint context verify
 Connecting with context "my-server"...
! Error connecting with context "my-server": context deadline exceeded
```

Tried with clean Ubuntu20 DO instance, the problem was with context initialisation on local machine, had to use this flag `-server-tls-skip-verify` and it worked fine after that.
